code/modules/reagents/chemistry/reagents/other_reagents.dm

```diff
@@ -2900,20 +2900,36 @@
 	metabolization_rate = 0.75 * REAGENTS_METABOLISM
 	material = /datum/material/hauntium
 	ph = 10
-	var/time_multiplier = 15 SECONDS
+	var/time_multiplier = 20 SECONDS
+	var/haunt_timer
 	chemical_flags = REAGENT_CAN_BE_SYNTHESIZED
 
 /datum/reagent/hauntium/expose_obj(obj/exposed_obj, volume) //gives 15 seconds of haunting effect for every unit of it that touches an object
 	. = ..()
-	exposed_obj.AddElement(/datum/element/haunted, 0)
-	addtimer(CALLBACK(exposed_obj, PROC_REF(_RemoveElement), list(/datum/element/haunted, 0)), volume * time_multiplier, TIMER_UNIQUE|TIMER_OVERRIDE)
+	exposed_obj.make_haunted(HAUNTIUM_REAGENT_TRAIT, "#f8f8ff")
+	haunt_timer = addtimer(CALLBACK(exposed_obj, TYPE_PROC_REF(/atom/movable/, remove_haunted), HAUNTIUM_REAGENT_TRAIT), volume * time_multiplier, TIMER_UNIQUE|TIMER_STOPPABLE)
+	if(haunt_timer) //if the timer for haunt removal is going and its splashed again, reset the timer.
+		deltimer(haunt_timer) //this looks kind of scuffed but every other way I tried it it just wouldn't work properly
+		haunt_timer = addtimer(CALLBACK(exposed_obj, TYPE_PROC_REF(/atom/movable/, remove_haunted), HAUNTIUM_REAGENT_TRAIT), volume * time_multiplier, TIMER_UNIQUE|TIMER_STOPPABLE)
```

Comment:
I'm a tad perplexed. While it may work on a one to one basis, it won't if the same reagent is splashed on several objects, or when several instances of the reagent splashed on the same object. At this point, let's keep it simple, even if it means hauntium smoke and foam won't be as strong. Also, I don't believe there's a need for a `time_multiplier` variable.
```suggestion
	if(HAS_TRAIT_FROM(exposed_obj, TRAIT_HAUNTED, HAUNTIUM_REAGENT_TRAIT)
		return
	exposed_obj.make_haunted(HAUNTIUM_REAGENT_TRAIT, "#f8f8ff")
	addtimer(CALLBACK(exposed_obj, TYPE_PROC_REF(/atom/movable/, remove_haunted), HAUNTIUM_REAGENT_TRAIT), volume * 20 SECONDS)
```